src/macrotools.jl

```diff
@@ -42,7 +52,7 @@ function splittypedef(ex)
         elseif item isa Symbol
             push!(d[:fields], (item, Any))
         else
-            push!(d[:constructors], item)
+            append!(d[:constructors], statements(item))
```

Comment:
There is functionality for removing blocks in MacroTools. I think you could just add [flatten](https://github.com/MikeInnes/MacroTools.jl/blob/e762757a65d6a4b45c90a942df2026a3ffb65207/src/utils.jl#L331) as another preprocessing step, like the striplines above.